doc/tech/approach.md

```diff
@@ -3,97 +3,226 @@
+
+SDNo controllers can request Underlay Networking encapsulation and mapping to be done by signaling to an SDNu controller. There are however today no standardized way for this signaling and by that there is a missing reference point and API description in this architecture. 
+
+Multiple instances of Container as a Service (CaaS) Virtualization Layers running on an Infrastructure as a Service (IaaS) Virtualization Layer could make use of the IaaS layer to handle the required Underlay Networking separation. In these cases, the IaaS Virtualisation Infrastructure Manager (VIM) could include a SDNu control interface enabling automation.
+
+> **Note:** The Reference Model describes a logical separation of SDNu and SDNo interfaces to clarify the separation of administrative domains where applicable. In real deployment cases an Operator can select to deploy a single SDN controller instance that implements all needed administrative domain separations or have separate SDN controllers for each administrative domain. A common deployment scenario today is to use a single SDN controller handling both Underlay and Overlay Networking which works on the implementation level when there is only one administrative domain that owns both the HW Infrastructure and the only  SW Virtualisation Infrastructure. The nature of a shared Underlay Network that shall ensure separation and be robust is that all code in the forwarding plane and in the control plane must be under the scrutiny and life cycle management of the HW Infrastructure Layer.
+One consequence of this is that the Reference Architectures must not model collapsed SDNo and SDNu controllers since they must stay unaware of other deployed implementations  running on the same HW Infrastructure.
+
+Hence, when Networking now gets introduced into the Reference Model, it is important that SDN also is considered and that the two now existing Reference Architectures are harmonised with the new common Networking Reference Model. In the absence of a Networking Reference Model, both Reference Architectures have mostly been specifying networking as if they where the only single SW Infrastructure on a dedicated HW resource pool.
+
+<a name="5.1.3.4"></a>
+#### 5.1.3.4 Programmable Networking Fabric concept
+
+The concept of a Programmable Networking Fabric pertains to the ability to have an effective forwarding pipeline (a.k.a. forwarding plane) that can be programmed and/or configured without any risk of disruption to the shared Underlay Networking that is involved with the reprogramming for the specific efficiency increase.
+
+The forwarding plane is distributed by nature and must be possible to implement both in switch elements and on SmartNICs (managed outside the reach of host SW) that both can be managed from a logically centralised control plane residing in the HW Infrastructure Layer.
```

Comment:
"...forwarding plane is **_physically_** distributed by nature..."

"...be managed from a logically centralised control plane..." - This control plane reference is ambiguous with respect to if the context is a forwarding control plane (i.e. programmable fabric (e.g. OF, P4, etc.)), or an SDN-type "configuration" control plane, meaning, for configuring attributes such as whether the encapsulation is applied at the switch interface or OVS/SmartNIC, tunnel IP assignment, etc., which would be common to centralized and distributed fabrics. Potentially need to add a third explicit control plane designation somewhat akin to an "SDNp" (p, as in the _programmable_ fabric's external forwarding control plane), to clarify the context.

Reply:
Implemented the physically.

I cannot fully understand your comment on a third control plane, but will try to make some comments anyhow since I think you are more discussing realizations then models.
The RM important concepts and separation are mostly pertaining to the separation of administrative domains i.e. who would have the rights and possibilities to do certain changes. How the internal structure of control software and switch + SmartNIC elements is an implementation choice.
Your references to an SDN-type control plane is actually a common implementation that is often used to handle a HW Infrastructure Layer with a single SW Virtualization instance that belongs to the same administrative domain as the HW Infrastructure instance do.